embrace-android-features/src/test/java/io/embrace/android/embracesdk/internal/capture/activity/UiLoadEventEmitterTest.kt

```diff
@@ -0,0 +1,339 @@
+package io.embrace.android.embracesdk.internal.capture.activity
+
+import android.app.Activity
+import android.os.Build
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.embrace.android.embracesdk.fakes.FakeClock
+import io.embrace.android.embracesdk.fakes.injection.FakeInitModule
+import io.embrace.android.embracesdk.internal.ClockTickingActivityLifecycleCallbacks
+import io.embrace.android.embracesdk.internal.ClockTickingActivityLifecycleCallbacks.Companion.POST_DURATION
+import io.embrace.android.embracesdk.internal.ClockTickingActivityLifecycleCallbacks.Companion.PRE_DURATION
+import io.embrace.android.embracesdk.internal.ClockTickingActivityLifecycleCallbacks.Companion.STATE_DURATION
+import io.embrace.android.embracesdk.internal.capture.activity.UiLoadEventEmitterTest.FakeOpenEvents.EventData
+import io.embrace.android.embracesdk.internal.utils.BuildVersionChecker
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.android.controller.ActivityController
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+internal class UiLoadEventEmitterTest {
+    private lateinit var clock: FakeClock
+    private lateinit var openEvents: FakeOpenEvents
+    private lateinit var eventEmitter: UiLoadEventEmitter
+    private lateinit var activityController: ActivityController<*>
+    private var startTimeMs: Long = 0L
+    private var instanceId = 0
+    private var activityName = ""
+
+    @Before
+    fun setUp() {
+        clock = FakeClock()
+        val initModule = FakeInitModule(clock = clock)
+        clock.tick(100L)
+        openEvents = FakeOpenEvents()
+        eventEmitter = UiLoadEventEmitter(
+            uiLoadEvents = openEvents,
+            clock = initModule.openTelemetryModule.openTelemetryClock,
+            versionChecker = BuildVersionChecker,
+        )
+        activityController = Robolectric.buildActivity(Activity::class.java)
+        RuntimeEnvironment.getApplication().registerActivityLifecycleCallbacks(
+            ClockTickingActivityLifecycleCallbacks(clock)
+        )
+        RuntimeEnvironment.getApplication().registerActivityLifecycleCallbacks(eventEmitter)
+        startTimeMs = clock.now()
+        instanceId = activityController.get().hashCode()
+        activityName = activityController.get().localClassName
+    }
+
+    @Config(sdk = [Build.VERSION_CODES.UPSIDE_DOWN_CAKE])
+    @Test
+    fun `check cold open event stages in U`() {
+        stepThroughActivityLifecycle()
+        openEvents.events.assertEventData(
+            listOf(
+                createEvent(
+                    stage = "create",
+                    timestampMs = startTimeMs + PRE_DURATION
+                ),
+                createEvent(
+                    stage = "createEnd",
```

Comment:
Are these values sent to the backend? I have similar feedback to the naming in `UiLoadEventEmitter` that this feels tricky to understand & I wouldn't like to communicate those names when debugging something on a call with a customer